Added summary report quickstart guide (#4246)
GoogleChrome/developer.chrome.com: site/en/docs/privacy-sandbox/summary-reports/design-decisions/index.md

---
layout: 'layouts/blog-post.njk'
title: Experiment with summary report design decisions
subhead: >
  Use Noise Lab to preview how noise affects your summary reports.
authors:
  - akashnadan
  - alexandrawhite
  - andreeacucu
  - maudn
  - zachmastromatto
date: 2022-11-09
updated: 2022-11-09
description: >
  Use Noise Lab to preview how noise affects your summary reports.
tags:
  - privacy
---

## About this document

> Readers will:

Reviewer: ```suggestion
When you read this article, you will:
```

Reviewer: A little more informal and welcoming.

work with their advertisers to develop noise management strategies.** To develop these strategies, DSPs and measurement providers need to make design decisions. These decisions revolve around one essential concept:

While the distribution _noise values_ are drawn from, absolutely speaking, only depends on two parameters⏤[epsilon and the contribution budget](https://docs.google.com/document/d/1bU0a_njpDcRd9vDR0AJjwJjrf3Or8vAzyfuK8JZDEfo/edit#heading=h.2994euq95tlo)⏤**you have a number of other controls at your disposal that will impact the
_signal-to-noise ratios_ of your output measurement data**.

While we expect an iterative process will lead to the best decisions, each variation on these decisions will
lead to a slightly different implementation—thus these decisions must be taken before writing each code iteration (and before running ads).

### Decision: Dimension granularity {: #dimension-granularity}

{% Aside %}
{% Details %}
{% DetailsSummary %}

#### Try it out in Noise Lab 🧪

{% endDetailsSummary %}

{% Aside %}
New to [Noise Lab](https://noise-lab.uc.r.appspot.com/)? [Take a quick tour](#quick-tour).
{% endAside %}

> 
> **Try it out:**

Reviewer: ```suggestion
```

Reviewer: Removing this repetition across the doc.

- Approach 2: one key structure with **granular** dimensions: _City_ x _Creative ID_ x _Product_ (out of 100 possible products)

_City_ is a more granular dimension than _Country_; _Creative ID_ is more granular
than _Campaign_; and _Product_ is more granular than _Product type_. Therefore,
Approach 2 will have a lower number of events (conversions) per bucket (= per
key) in its summary report output than Approach 1. Given that the noise added to
the output is agnostic to the number of events in the bucket, measurement data
in summary reports will be more noisy with Approach 2. **For each advertiser, experiment with various
granularity tradeoffs in the key's design in order to have maximum utility in
the results.**

### Decision: Key structures {: #key-structures}

{% Aside %}
{% Details %}
{% DetailsSummary %}

#### Try it out in Noise Lab 🧪

{% endDetailsSummary %}

{% Aside %}
New to [Noise Lab](https://noise-lab.uc.r.appspot.com/)? [Take a quick tour](#quick-tour).
{% endAside %}

In the **Simple** mode, the default key structure is used. In the **Advanced**
mode, you can experiment with different key structures. Some example dimensions
are included; you can also modify these.

> 
> **Try it out:**

Reviewer: ```suggestion
```

Strategy A is simpler—but you may need to roll up (sum) the noisy summary values included summary reports to access certain insights. By summing up these values, you're also summing the noise.
With Strategy B, summary values exposed in the summary
reports may already give you the information you need. This means that Strategy B
will likely lead to better signal-to-noise ratios than Strategy A. However, the
noise may already be acceptable with Strategy A, so you may still decide to favor
Strategy A for simplicity.
[Learn more in the detailed example outlining these two strategies](https://docs.google.com/document/d/1bU0a_njpDcRd9vDR0AJjwJjrf3Or8vAzyfuK8JZDEfo/edit#heading=h.i6ovl4dm1npa).

Key management is a deep topic. A number of elaborate techniques can be
considered to improve signal-to-noise ratios. One is described in [Advanced key
management](#key-management-advanced).

### Decision: Batching frequency {: #batching-frequency}

{% Aside %}
{% Details %}
{% DetailsSummary %}

#### Try it out in Noise Lab 🧪

{% endDetailsSummary %}

{% Aside %}
New to [Noise Lab](https://noise-lab.uc.r.appspot.com/)? [Take a quick tour](#quick-tour).
{% endAside %}

> **Try it out:**
> 

Reviewer: ```suggestion
```

have fewer conversion events included than the same report with a less frequent
aggregation schedule (e.g. each week). As a result, the hourly report will have
a higher signal-to-noise ratio than the weekly report, all else being equal. **Experiment with reporting requirements at various frequencies, and assess
signal-to-noise ratios for each.**

Learn more in
[Batching](https://docs.google.com/document/d/1bU0a_njpDcRd9vDR0AJjwJjrf3Or8vAzyfuK8JZDEfo/edit#heading=h.pef51jskrx06)
and [Aggregating over longer time periods](https://docs.google.com/document/d/1bU0a_njpDcRd9vDR0AJjwJjrf3Or8vAzyfuK8JZDEfo/edit#heading=h.pef51jskrx06).

### Decision: Campaign variables that affect attributable conversions {: #campaign-variables}

{% Aside %}
{% Details %}
{% DetailsSummary %}

#### Try it out in Noise Lab 🧪

{% endDetailsSummary %}

{% Aside %}
New to [Noise Lab](https://noise-lab.uc.r.appspot.com/)? [Take a quick tour](#quick-tour).
{% endAside %}

While this can be difficult to predict and can have significant variations in
addition to seasonality effects, try and estimate the number of daily
single-touch attributable conversions to the closest power of 10: 10, 100,
1,000, or 10,000.

> **Try it out:**

Reviewer: ```suggestion
```

variables, such as ad budget and ad targeting. For example, you would expect a
higher number of attributed conversions for a $10M ad campaign vs. a $10K ad
campaign, all else being equal.

Things to consider:

- Assess attributed conversions against a single-touch, same device
  attribution model, as these are within the scope of summary reports
  collected with the Attribution Reporting API.
- Consider both a worst-case scenario count and a best-case scenario count
  for attributed conversions. For example, all else being equal, consider the
  minimum and maximum possible campaign budgets for an advertiser, then
  project attributable conversions for both outcomes as inputs into your
  simulation.
- If you're considering using the
  [Android Privacy Sandbox](https://developer.android.com/design-for-safety/privacy-sandbox),
  consider cross-platform attributed conversions in the calculation.

### Decision: Using scaling {: #scaling}

{% Aside %}
{% Details %}
{% DetailsSummary %}

#### Try it out in Noise Lab 🧪

{% endDetailsSummary %}

{% Aside %}
New to [Noise Lab](https://noise-lab.uc.r.appspot.com/)? [Take a quick tour](#quick-tour).
{% endAside %}

> **Try it out:**

Reviewer: ```suggestion
```

    setup, you'd want to implement your own calculation for scaling factors.
1.  Click **Simulate**.
1.  Observe that the noise ratios are now lower (better) in this second
    simulation. This is because you're using scaling.

{% endDetails %}
{% endAside %}

Given the [core design principle](#core), the noise added is
a function of the contribution budget.

Therefore, to increase signal-to-noise ratios, you can decide to transform
values collected during a conversion event by [scaling](https://docs.google.com/document/d/1bU0a_njpDcRd9vDR0AJjwJjrf3Or8vAzyfuK8JZDEfo/edit#heading=h.683u7t2q1xk2) them against the
contribution budget (and de-scaling them after aggregation). **Use scaling to increase signal-to-noise ratios**.

### Decision: Number of measurement goals, and privacy budget split {: #goals-and-budget}

This relates to Scaling; make sure to read [Using
scaling](#scaling).

{% Aside %}
{% Details %}
{% DetailsSummary %}

#### Try it out in Noise Lab 🧪

{% endDetailsSummary %}

{% Aside %}
New to [Noise Lab](https://noise-lab.uc.r.appspot.com/)? [Take a quick tour](#quick-tour).
{% endAside %}

A measurement goal is a distinct data point collected in conversion events.

> **Try it out:**

Reviewer: ```suggestion
```

   this, it needs "more space to scale": it would be ideal to assign more
   contribution budget to **Measurement goal 1** than **Measurement goal 2**, so that
   it can be scaled up more efficiently (see Scaling), and hence
1. Assign 70% of the budget to **Measurement goal 1**. Assign 30% to **Measurement
   goal 2**.
1. Click **Simulate**.
1. Observe the noise ratios of the resulting simulation. For purchase
   value, noise ratios are now notably lower (better) than for the previous
   simulation. For purchase count, they're roughly unchanged.
1. Keep tweaking the budget split across metrics. Observe how this impacts
   noise.

Note that you can set your own custom measurement goals with the
**Add**/**Remove**/**Reset** buttons.

{% endDetails %}
{% endAside %}

If you measure one data point (measurement goal) on a conversion event, such as
conversion count, that data point can get all the contribution budget (65536). If you set multiple measurement goals on a conversion event,
such as conversion count and purchase value, then those data points will need to
share the contribution budget. This means you have less leeway to scale up your

> values. **Therefore, the more measurement goals you have, the lower signal-to-noise ratios are likely to be (higher noise)**.

Reviewer: ```suggestion
values. 

**Therefore, the more measurement goals you have, the lower signal-to-noise ratios
are likely to be (higher noise)**.
```

Another decision to make regarding measurement goals is the budget split. If you split the contribution budget equally across two data points, each data point gets a
budget of 65536/2 = 32768. This may or may not be optimal depending on the
maximum possible value for each data point. For example, if you're measuring
purchase count that has a maximum value of 1, and purchase value with a
minimum of 1 and a maximum of 120, the purchase value would benefit from having
"more space" to be scaled up—that is, to be given a larger proportion of the

> contribution budget. **See if some measurement goals need to be prioritized over
> others in relation to the impact of noise.**

Reviewer: ```suggestion
contribution budget. **You'll see if some measurement goals should be prioritized over
others in relation to the impact of noise.**
```

### Decision: Outlier management {: #outlier-management}

{% Aside %}
{% Details %}
{% DetailsSummary %}

#### Try it out in Noise Lab 🧪

{% endDetailsSummary %}

{% Aside %}
New to [Noise Lab](https://noise-lab.uc.r.appspot.com/)? [Take a quick tour](#quick-tour).
{% endAside %}

A measurement goal is a distinct data point collected in conversion events.

> **Try it out:**

Reviewer: ```suggestion
```

Reviewer: The section heading is "Try it out", so no need for repeat!

#### Simple mode

- With Simple mode, you control **Parameters** (found on the left-hand
  side, or _#2. in the screenshot below_) such as Epsilon, and see how they impact noise.
- Each parameter has a tooltip (a `?` button). Click these to see an
  explanation of each parameter (#_3. in the screenshot below_)
- To start, click the "Simulate" button and observe what the output looks
  like (#_4. in the screenshot below_)
- In the **Output** section you can see a variety of details. Some
  elements have a `?` next to it. Take the time to click each `?` to see an
  explanation of the various pieces of information.
  - Within the **Output** section, click on the **Details** toggle
    if you want to see an expanded version of the table (#_5. in the screenshot below)_
  - Below each data table in the output section, there is an option
    to download the table for offline usage. Additionally, in the bottom
    right corner there is an option to download all data tables (#_6. in
    the screenshot below_)
- Test different settings for the parameters in the **Parameters** section
  and click **Simulate** to see how they impact the output:

<figure>

> {% Img src="image/EvKqfXr4ercavLaApW41d8uGXN72/8o20EoPwoN2fxoNHPg82.jpg", alt="Screenshot: Noise Lab interface for Simple mode", width="800", height="450" %}

Reviewer: ```suggestion
{% Img src="image/EvKqfXr4ercavLaApW41d8uGXN72/8o20EoPwoN2fxoNHPg82.jpg", alt="Noise Lab interface for Simple mode", width="800", height="450" %}
```

  <figcaption>
    Noise Lab interface for Simple mode.
  </figcaption>
</figure>

#### Advanced mode

- In Advanced mode, you have more control over the **Parameters**. You
  can add custom Measurement Goals and Dimensions (#_1. and #2. in the screenshot
  below_)
- Scroll further down in the **Parameters** section and see the **Key
  Strategy** option. This can be used to test out different key structures
  (#_3. in the screenshot below_)
  - To test out different Key Structures, switch **Key Strategy** to "B"
  - Input the number of different key structures you want to use
    (default is set to "2")
  - Click **Generate Key Structures**
  - You will see options to specify your key structures by clicking
    the checkboxes next to the keys you want to include for each key structure
  - Click **Simulate** to see the output.

<figure>

> {% Img src="image/EvKqfXr4ercavLaApW41d8uGXN72/5qH8KUvKhhW4vbUbNzbT.jpg", alt="Screenshot: Noise Lab interface for Advanced mode", width="800", height="450" %}

Reviewer: ```suggestion
{% Img
  src="image/EvKqfXr4ercavLaApW41d8uGXN72/5qH8KUvKhhW4vbUbNzbT.jpg",
  alt="Advanced mode offers controls for measurement goals and dimensions to track, highlighted in the sidebar.",
  width="800", height="450"
%}
```

  <figcaption>
    Noise Lab interface for Advanced mode.
  </figcaption>
</figure>

<figure>

> {% Img src="image/EvKqfXr4ercavLaApW41d8uGXN72/DfUCCLzvay9SgExqUMc0.jpg", alt="Screenshot: Noise Lab interface for Advanced mode", width="800", height="450" %}

Reviewer: ```suggestion
{% Img
  src="image/EvKqfXr4ercavLaApW41d8uGXN72/DfUCCLzvay9SgExqUMc0.jpg",
  alt="Advanced mode also a Key strategy option in the Parameters section of the sidebar.",
  width="800", height="450"
%}
```
